teps/0071-custom-task-sdk.md

```diff
@@ -0,0 +1,220 @@
+---
+status: proposed
+title: Custom Task SDK
+creation-date: '2021-06-15'
+last-updated: '2021-06-15'
+authors:
+- '@ScrapCodes'
+---
+
+# TEP-0071: Custom Task SDK
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+    - [Use Cases (optional)](#use-cases-optional)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+    - [Notes/Caveats (optional)](#notescaveats-optional)
+    - [Risks and Mitigations](#risks-and-mitigations)
+    - [User Experience (optional)](#user-experience-optional)
+    - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+- [Test Plan](#test-plan)
+- [Design Evaluation](#design-evaluation)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [Implementation Pull request(s)](#implementation-pull-request-s)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+A Custom Task SDK will make the work of custom task author easier and will offer
+standard way to implement custom task controllers. While adhering to current
+separation of concern between a custom task controller's role and tektoncd owned
+controller's role, it should make custom tasks easier to manage and reason by
+tektoncd.
+
+## Motivation
+
+1. Currently, a custom task author has limited ways to know what is a standard way
+    of building his custom task. His best bet is to explore one of the existing 
```

Comment:
nit: possible to use gender neutral "they/them/their" instead of "he/him/his"? (same applies in other parts of this doc as well)

(im planning to put some recommendations around this into our standards docs eventually :))